docs: pr review process (#25899)
langchain-ai/langchain: docs/docs/contributing/review_process.mdx

# Review Process

## Overview

> This document outlines the process and priorities used by the LangChain team for reviewing and managing pull requests (PRs). The primary objective of this process is to enhance the LangChain development experience for users. This is achieved by prioritizing PRs that positively impact the user experience. Below is the current process we follow.

Reviewer: nit
```suggestion
This document outlines the process used by the LangChain maintainers for reviewing pull requests (PRs). The primary objective of this process is to enhance the LangChain developer experience.
```

  - Requires a maintainer to categorize it into one of the other statuses.

- **Needs Support**:
  - PRs that require community feedback or additional input before moving forward.
  - Automatically promoted to the backlog if it receives 5 upvotes.
  - An auto-comment is generated when this status is applied, explaining the flow and the upvote requirement.
  - If the PR remains in this status for 25 days, it will be marked as “stale” via auto-comment.
  - PRs will be auto-closed after 30 days if no further action is taken.

- **In Review**:
  - PRs that are actively under review by our team.
  - These are regularly reviewed and monitored.

**Note:** A PR may only have one status at a time.

**Note:** You may notice 3 additional statuses of Done, Closed, and Internal that
are external to this lifecycle. Done and Closed PRs have been merged or closed,
respectively. Internal is for PRs submitted by core maintainers, and these PRs are owned
by the submitter.

## Review Guidelines

1. **PRs that touch /libs/core**:
   - PRs that directly impact core code and are likely to affect end users.
   - **Triage Guideline**: most PRs should either go straight to `In Review` or closed.

>    - These are reviewed most aggressively, and may also be closed more aggressively if necessary.

Author: accepting and adding some wording around closing. I think it's worth calling out that PRs should avoid touching core if possible

   - **Triage Guideline**: most PRs should either go straight to `In Review` or closed.
   - These are reviewed and closed aggressively, similar to core PRs.
   - New feature requests should be discussed with the core maintainer team beforehand in an issue.

3. **Partner PRs**:
   - PRs involving partner packages.
   - **Triage Guideline**: most PRs should either go straight to `In Review` or closed.
   - The review may be conducted by our team or handed off to the partner's development team, depending on the PR's content.
   - We maintain communication lines with most partner dev teams to facilitate this process.

4. **Community PRs**:
   - Most community PRs will get an initial status of "needs support"
   - **Triage Guideline**: most PRs should go to `Needs support`. Bugfixes on high-traffic integrations should go straight to `In review`.
   - **Triage Guideline**: all new features and integrations should go to `Needs support` and will be closed if they do not get enough support (measured by upvotes or comments).
   - PRs in the `Needs Support` status for 20 days are marked as “stale” and will be closed after 30 days if no action is taken.

> 5. **Documentation PRs**:

Reviewer: doesn't say much about how we prioritize them. maybe something like
- corrections / fixes
- improving docs coverage
- refactoring docs

   - We strive to standardize documentation formats to streamline the review process.
   - CI jobs run against documentation to ensure adherence to standards, automating much of the review.

## How to see a PR's status

See screenshot:

![PR Status](/img/review_process_status.png)

*To see the status of all open PRs, please visit the [LangChain Project Board](https://github.com/orgs/langchain-ai/projects/12/views/2).*

## Review Prioritization

> Our goal is to provide the best possible development experience by focusing on:
> 
> - Bug-free core abstractions.
> - Bug-free integrations and implementations.
> - New integrations and implementations that align with industry developments.

Reviewer: ```suggestion
Our goal is to provide the best possible development experience by focusing on making software that:

- Works: Works as intended (is bug-free).
- Is useful: Improves LLM app development with components that work off-the-shelf and runtimes that simplify app building.
- Is easy: Is intuitive to use and well-documented.
```

We believe this process reflects our priorities and are open to feedback if you feel it does not.

## Github Discussion

> We welcome your feedback on this process. Please feel free to add a comment in the pinned Github Discussion on "Review Process".

Reviewer: should we link

Author: Going to post as an announcement after this deploys, and then link in second small pr
